daemon/attach.go

```diff
@@ -113,132 +113,90 @@ func (daemon *Daemon) ContainerAttach(job *engine.Job) engine.Status {
 
 func (daemon *Daemon) attach(streamConfig *StreamConfig, openStdin, stdinOnce, tty bool, stdin io.ReadCloser, stdout io.Writer, stderr io.Writer) chan error {
 	var (
-		cStdout, cStderr io.ReadCloser
-		nJobs            int
-		errors           = make(chan error, 3)
+		cStdout = streamConfig.StdoutPipe()
+		cStderr = streamConfig.StderrPipe()
+		cStdin  io.WriteCloser
+		nJobs   = 3 // stdin, stdout, stderr
+		errors  = make(chan error, nJobs)
 	)
 
+	if stdin != nil && openStdin {
+		cStdin = streamConfig.StdinPipe()
+	}
+
+	closePipes := func() {
```

Comment:
I found this little confusing. Can we just do this where it needed, without closure creation.